header-rewrite-filter/header_processor.cc

```diff
@@ -23,24 +28,40 @@ namespace HeaderRewriteFilter {
         try {
             std::vector<std::string> vals;
             for(auto it = operation_expression.begin() + 3; it != operation_expression.end(); ++it) {
+                if (*it == "if") { // condition found
```

Comment:
seems like this is duplicated between set-header / set-path, could you extract it into a helper?